redis/client.py

```diff
@@ -938,6 +938,7 @@ def __init__(
         username=None,
         retry=None,
         redis_connect_func=None,
+        credentials_provider=None,
```

Comment:
Can this be credential_provider - there aren't many.
Also, a type hint please on all new items